### DISCUSSION TITLE
Import BRouter GeoJSON routes

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#454] [Q&A] [answered]
I like to use the BRouter Web Client for route planning because I can set the BRouter there just like on my smartphone.
Exporting with GPX is no problem either.
The web client also has a format for exporting: "GeoJSON".
This format is probably something different from the JSON format at CRUISER?

**Comment by devemux86** [comment]
BRouter GeoJSON is an alternative export route format of  BRouter website.
If you check it, you will see that it contains more data than the GPX format.

> This format is probably something different from the JSON format at CRUISER?

They are different formats.

Cruiser JSON format is an offline, compact format that has no limits and can hold any possible data.

**Comment by devemux86** [comment]
If BRouter GeoJSON format can contain offline all routing data,
we could use it in Cruiser for offline import of BRouter website routes?

**Reply by Leeb64** [reply]
This is what I found:
GeoJSON is an open standard spatial data exchange format for representing simple geographic features and their nonspatial attributes. GeoJSON is based on JSON (JavaScript Object Notation) and is a format for encoding various structures of geographical data.

**Comment by devemux86** [comment]
> I like to use the BRouter Web Client for route planning because I can set the BRouter there just like on my smartphone.

@Leeb64 :+1: 

You can also try the GPX navigation format that can be exported by BRouter website.
It is an offline format that can be imported directly in Cruiser. See instructions here:

- https://github.com/devemux86/cruiser/discussions/272#discussioncomment-7599601

Note: GPX navigation format by BRouter website, does not contains speed limits.

**Comment by devemux86** [comment]
Cruiser now imports [BRouter website](https://brouter.de/brouter-web/) GeoJSON routes.

- Enable **turn instructions** in profile options

- Export in GeoJSON format (check **Waypoints**)

- Import the GeoJSON in Cruiser (**offline**)

![BRouter](https://github.com/devemux86/cruiser/assets/3484020/ec4b3906-3b8a-4b9d-aeba-0f72d494a408)

**Reply by devemux86** [reply]
Or export to **GPX navigation format**, which can already be imported (**offline**) in Cruiser. Instructions are here:
- https://github.com/devemux86/cruiser/discussions/272#discussioncomment-7599601

**Reply by devemux86** [reply]
BRouter website does not separate the intermediate points to via or shaping.

How should we import the intermediate points of BRouter website in Cruiser?
As via or shaping points?

**Reply by Leeb64** [reply]
Thats better!:     
https://brouter.de/essbee/#map=10/48.8408/7.7419/OpenTopoMap
You can upload your own profiles.

**Reply by Leeb64** [reply]
I think VIA Points would be better....

**Reply by devemux86** [reply]
@Leeb64 we can use our custom profiles in all BRouter websites in the same way:

- https://github.com/devemux86/cruiser/discussions/272

**Reply by devemux86** [reply]
There is also the BRouter website: https://bikerouter.de

>  You can upload your own profiles.

@Leeb64 indeed https://brouter.de/essbee/ also has a profiles menu to upload custom profiles.

**Comment by devemux86** [comment]
Prefer the GeoJSON format in BRouter **website** exports.
It contains much more data than GPX, such as speed limits, road surface, etc.

So it will also be more useful in new features in the future.

**Comment by devemux86** [comment]
Import BRouter website GeoJSON routes (**offline**) is available for testing in:
- https://github.com/devemux86/cruiser/discussions/451

---

For instructions, please see above:
- https://github.com/devemux86/cruiser/discussions/454#discussioncomment-8570130

**Comment by devemux86** [comment, marked as answer]
Import BRouter GeoJSON routes is available in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
